### DISCUSSION TITLE
Does Remark Math Parser Require Ending `$$` to be on its Own Line?

### DISCUSSION THREAD

**Discussion by pjkaufman** [remarkjs/remark#1069] [Q&A] [answered]
When parsing the AST for math, it looks like it is considering all text after the `$$` to be still a part of the math block when it should stop at the ending math block symbol. Here is the online AST parsing example: https://astexplorer.net/#/gist/f84501a2b986e7656461e1c51e408abb/1433512c1fc0edf26c7eb64af46b88d8d2ccd353

I have not tried this out in a JS package yet, but essentially the gist of the problem is as follows.

The input is 
``` plaintext
$$
\displaylines{ a=b,\\
c=d}
$$

test

$$
\displaylines{ a=b,\\
c=d
}$$

test

test

test
```

The first math block is correctly listed as 
``` plaintext
$$
\displaylines{ a=b,\\
c=d}
$$
```

However the second math block is said to be 
```plaintext
$$
\displaylines{ a=b,\\
c=d
}$$

test

test

test
```
That seems wrong to me, but I wanted to verify that this is the intended functionality of the remark-math. I plan to test out the pure JS npm package on this soon to see if it also has this issue since I do see this when I use remark-math as a part of a linter plugin for Obsidian.

Any feedback on what could be causing this would be great!

I had a previous discussion where it was said that the math block indicators work kind of like code blocks in markdown and wanted to make sure that the syntax does require the dollar signs on their own line before I consider this to be an actual issue with the parser: https://github.com/orgs/remarkjs/discussions/1056

**Comment by ChristianMurphy** [comment, marked as answer]
Yes, for a block `$$` needs to be on it's own line.

---

Again, math isn't fully standardized (as noted in https://github.com/orgs/remarkjs/discussions/1056), but we can use code blocks, which are standardized, as a proxy.
Take the following code block as an example:

````markdown
```
code block
```

end block

```
{
code block
}```

still a code block
```
close code block
````

rendered:

---


```
code block
```

end block

```
{
code block
}```

still a code block
```
close code block

---

It has the same behavior as math blocks.
I believe the `remark-math` implementation/behavior as it stand currently, is correct and appropriate.

---

Another thing work noting.
GitHub also supports math, at some point, it may be worth making sure implementations align.
However, GitHub's own implementation is still in an early stage, and is not standardized as part of GFM yet.
https://github.github.com/gfm/

**Reply by pjkaufman** [reply]
Thanks for the update. I was not sure if by saying it worked like code blocks that meant that the dollar signs needed to be on their own lines or if they were on the same line like inline math, it would work just fine.
